src/expr.rs

```diff
@@ -1522,10 +1525,15 @@ pub fn rewrite_assign_rhs<S: Into<String>>(context: &RewriteContext,
                                            offset: Indent)
                                            -> Option<String> {
     let mut result = lhs.into();
-
+    let last_line_width = last_line_width(&result) -
+                          if result.contains('\n') {
```

Comment:
It is, yes. The current heuristic breaks the line when the rhs doesn't fit the same line in its entirety.

-----Original Message-----
From: "Nick Cameron" <notifications@github.com>
Sent: ‎25-‎10-‎2015 18:37
To: "nrc/rustfmt" <rustfmt@noreply.github.com>
Cc: "Marcus Klaas de Vries" <mail@marcusklaas.nl>
Subject: Re: [rustfmt] Address some issues with multiline patterns in letstatements (#531)

In src/expr.rs:
> @@ -1522,10 +1525,15 @@ pub fn rewrite_assign_rhs<S: Into<String>>(context: &RewriteContext,
>                                             offset: Indent)
>                                             -> Option<String> {
>      let mut result = lhs.into();
> -
> +    let last_line_width = last_line_width(&result) -
> +                          if result.contains('\n') {
Is rustfmt splitting the line here? Looks like we shouldn't if it is.
—
Reply to this email directly or view it on GitHub.